crates/iceberg/src/arrow/reader.rs

```diff
@@ -219,10 +223,23 @@ impl ArrowReader {
             initial_stream_builder
         };
 
+        // Build the metadata transformer from the projected field IDs
+        // This identifies reserved fields (like _file) and creates transformations for them
+        let metadata_transformer_builder =
+            MetadataTransformer::builder(task.project_field_ids.clone())
+                .with_file_path(task.data_file_path.clone());
+
+        // Get the field IDs without virtual fields for Parquet projection
+        let project_field_ids_without_virtual =
+            metadata_transformer_builder.project_field_ids_without_virtual();
```

Comment:
It feels a bit disconnected that one has to construct `metadata_transformer_builder` to get `project_field_ids_without_virtual`. But I do agree it should belong to that module/struct. Perhaps it can be a function without taking `self` parameter? Instead you could pass `task.project_field_ids` to it?